base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -133,21 +134,28 @@ impl<B: BlockchainBackend> Clone for Validators<B> {
 /// us to keep the reading and writing API extremely simple. Extending the types of data that the back ends can handle
 /// will entail adding to those enums, and the back ends, while this trait can remain unchanged.
 pub trait BlockchainBackend: Send + Sync {
-    /// Commit the transaction given to the backend. If there is an error, the transaction must be rolled back, and
-    /// the error condition returned. On success, every operation in the transaction will have been committed, and
+    /// Adds a block to the orphan database. On success, every operation in the transaction will have been committed,
+    /// and the function will return `Ok(())`.
+    fn add_orphan_block(&mut self, block: Block) -> Result<(), ChainStorageError>;
+    /// This function moves the block from the orphan pool to the main db. On success, every operation in the
+    /// transaction will have been committed, and the function will return `Ok(())`.
+    fn add_block(&mut self, block_hash: HashOutput) -> Result<(), ChainStorageError>;
+    // rewinds the database to the specified height. On success, every operation in the transaction will have been
+    // committed, and
     /// the function will return `Ok(())`.
-    fn write(&mut self, tx: DbTransaction) -> Result<(), ChainStorageError>;
+    fn rewind_to_height(&mut self, height: u64) -> Result<(), ChainStorageError>;
+    /// Commit the transaction given to the backend. This will create a db write transaction. This inly for syncing. On
+    /// success, every operation in the transaction will have been committed, and the function will return `Ok(())`.
+    fn write_sync(&mut self, tx: DbTransaction) -> Result<(), ChainStorageError>;
     /// Fetch a value from the back end corresponding to the given key. If the value is not found, `get` must return
     /// `Ok(None)`. It should only error if there is an access or integrity issue with the underlying back end.
     fn fetch(&self, key: &DbKey) -> Result<Option<DbValue>, ChainStorageError>;
     /// Checks to see whether the given key exists in the back end. This function should only fail if there is an
     /// access or integrity issue with the back end.
     fn contains(&self, key: &DbKey) -> Result<bool, ChainStorageError>;
-    /// Fetches the merklish root for the MMR tree identified by the key. This function should only fail if there is an
-    /// access or integrity issue with the back end.
-    fn fetch_mmr_root(&self, tree: MmrTree) -> Result<HashOutput, ChainStorageError>;
+    // DELETE? fn fetch_mmr_root(&self, tree: MmrTree) -> Result<HashOutput, ChainStorageError>;
```

Comment:
I believe we still need this

Reply:
This is never called in code, only in unit tests.